### PR TITLE
add chat tool calling (invoke only)

### PR DESCRIPTION
**Pull request by mattf**
```
from langchain_core.pydantic_v1 import Field
from langchain_core.tools import tool
from langchain_nvidia_ai_endpoints import ChatNVIDIA

@tool
def xxyyzz(
    a: int = Field(..., description="First number"),
    b: int = Field(..., description="Second number"),
) -> int:
    """xxyyzz two numbers"""
    return (a**b) % (b - a)

llm = ChatNVIDIA(model=tool_model, **mode).bind_tools(tools=[xxyyzz])
response = llm.invoke("What is 11 xxyyzz 3?", tool_choice="required")
print(response.tool_calls)
```
```
[{'name': 'xxyyzz', 'args': {'a': 11, 'b': 3}, 'id': 'chatcmpl-tool-f80cd197c04e41c6be7a86bb20434e01'}]
```

**Note:** `invoke` now returns an `AIMessage` when a tool call is returned and a `ChatMessage` otherwise. The old behavior was to always return a `ChatMessage`. This new behavior is not technically API breaking because `invoke` declares it returns a `BaseMessage` and both `AIMessage` & `ChatMessage` are children of `BaseMessage`. However, `AIMessage` and `ChatMessage` are siblings and not compatible. `ChatMessage` has a `role` property that is not present on either `BaseMessage` or `AIMessage`. A user who expected `role` in the response will be broken by this change.

### CODE REVIEW

File: libs/ai-endpoints/tests/integration_tests/test_bind_tools.py [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+
+import pytest
+from langchain_core.messages import AIMessage, ChatMessage
+from langchain_core.pydantic_v1 import Field
+from langchain_core.tools import tool
+
+from langchain_nvidia_ai_endpoints import ChatNVIDIA
+
+#
+# ways to specity tools:
```
Comment (dglogo) [inline libs/ai-endpoints/tests/integration_tests/test_bind_tools.py:13]: nit: specify

File: libs/ai-endpoints/langchain_nvidia_ai_endpoints/_common.py [hunk 2]
```diff
@@ -368,13 +368,17 @@ def _aggregate_msgs(self, msg_list: Sequence[dict]) -> Tuple[dict, bool]:
...
         is_stopped = False
         for msg in msg_list:
             usage_holder = msg.get("usage", {})  ####
             if "choices" in msg:
                 ## Tease out ['choices'][0]...['delta'/'message']
                 msg = msg.get("choices", [{}])[0]
-                is_stopped = msg.get("finish_reason", "") == "stop"
+                # todo: this meeds to be fixed, the fact we only
+                #       use the first choice breaks the interface
+                finish_reason_holder = msg.get("finish_reason", None)
```
Comment (raspawar) [inline libs/ai-endpoints/langchain_nvidia_ai_endpoints/_common.py:380]: I guess this variable is not necessary: finish_reason_holder

is_stopped = msg.get("finish_reason") == "stop" should do?
and the if finish_reason_holder: can use is_stopped?

**Comment by raspawar** [comment]
Reviewed code impl, looks great!

File: libs/ai-endpoints/langchain_nvidia_ai_endpoints/_common.py (hunk 2 above)
Comment (mattf) [reply libs/ai-endpoints/langchain_nvidia_ai_endpoints/_common.py:380]: it's also used in https://github.com/langchain-ai/langchain-nvidia/pull/72/files#diff-6b48e6b0775169741d8200a861c8bbab23256e2d1986caa9cd56d74624993ee4R399

**Pull request closed without merging**